include/hal/library/memlib.h

```diff
@@ -16,24 +16,32 @@
 #define __BASE_MEMORY_LIB__
 
 /**
- * Copies a source buffer to a destination buffer, and returns the destination buffer.
+ * Copies bytes from a source buffer to a destination buffer.
  *
- * This function copies length bytes from source_buffer to destination_buffer, and returns
- * destination_buffer.  The implementation must be reentrant, and it must handle the case
- * where source_buffer overlaps destination_buffer.
+ * This function copies "len" bytes from "src_buf" to "dst_buf".
  *
- * If length is greater than (MAX_ADDRESS - destination_buffer + 1), then ASSERT().
- * If length is greater than (MAX_ADDRESS - source_buffer + 1), then ASSERT().
+ * Asserts and returns a non-zero value if any of the following are true:
+ *   1) ("src_buf" or "dst_buf" are NULL) and "len" is greater 0.
+ *   2) "src_buf" and "dst_buf" overlap.
+ *   3) "len" is greater than "dst_len".
+ *   4) "len" or "dst_len" is greater than (MAX_ADDRESS - "dst_buf" + 1).
+ *   5) "len" or "dst_len" is greater than (MAX_ADDRESS - "src_buf" + 1).
+ *   6) "len" or "dst_len" is greater than (SIZE_MAX >> 1).
  *
- * @param  destination_buffer   The pointer to the destination buffer of the memory copy.
- * @param  source_buffer        The pointer to the source buffer of the memory copy.
- * @param  length              The number of bytes to copy from source_buffer to destination_buffer.
+ * In case of error, the "dst_buf" is left unmodifed. This behavior is different
+ * than memcopy_s, which zeros the "dst_buf" if "dst_buf" is valid.
```

Comment:
I would say:
This behavior is different with C11 memcpy_s, which zeros the "dst_buf" if there is runtime violation.

Reply:
ok.